torchaudio/datasets/vctk.py

```diff
@@ -118,20 +118,23 @@ def __getitem__(self, index):
         Returns:
             tuple: (image, target) where target is index of the target class.
         """
-        if self.cached_pt != index // self.chunk_size:
-            self.cached_pt = int(index // self.chunk_size)
-            self.data, self.labels = torch.load(os.path.join(
-                self.root, self.processed_folder, "vctk_{:04d}.pt".format(self.cached_pt)))
-        index = index % self.chunk_size
-        audio, target = self.data[index], self.labels[index]
-
-        if self.transform is not None:
-            audio = self.transform(audio)
-
-        if self.target_transform is not None:
-            target = self.target_transform(target)
-
-        return audio, target
+        try :
+            if self.cached_pt != index // self.chunk_size:
+                self.cached_pt = int(index // self.chunk_size)
+                self.data, self.labels = torch.load(os.path.join(
+                    self.root, self.processed_folder, "vctk_{:04d}.pt".format(self.cached_pt)))
+            index = index % self.chunk_size
+            audio, target = self.data[index], self.labels[index]
+
+            if self.transform is not None:
+                audio = self.transform(audio)
+
+            if self.target_transform is not None:
+                target = self.target_transform(target)
+
+            return audio, target
+        except IndexError :
+            return self.__getitem__()
```

Comment:
i'm not sure I understand this change. How does `self.__getitem__()` work at all?

Reply:
`return self.__getitem__()` skips the current sample and returns the next one. For our usage, we returned an integer value and were checking in the training loop. However, I thought this would generalize better.

Reply:
I don't think this is the right solution. The correctness of the index should be checked instead. maybe close?